warehouse/models/intermediate/payments/int_payments__refunds_deduped.sql

```diff
@@ -0,0 +1,156 @@
+        SAFE_CAST(NULL AS STRING) AS settlement_status,
+        SAFE_CAST(NULL AS DATE) AS settlement_status_time,
+        SAFE_CAST(NULL AS STRING) AS settlement_reason_code,
+        SAFE_CAST(NULL AS STRING) AS settlement_response_text,
+
+        _line_number,
+        currency_code,
+        instance,
+        extract_filename,
+        ts,
+        littlepay_export_ts,
+        littlepay_export_date,
+        _content_hash,
+        _key,
+        _payments_key,
+        'micropayments' AS source_table
+
+    FROM {{ ref('stg_littlepay__micropayments') }}
+    WHERE type = 'CREDIT'
+),
+
+refunds_table_refunds AS (
+    SELECT
+
+        aggregation_id,
+        micropayment_id,
+        participant_id,
+        customer_id,
+        refund_amount,
+        transaction_date,
+        refund_id,
+        settlement_id,
+        retrieval_reference_number,
+        transaction_amount,
+        proposed_amount,
+        status,
+        initiator,
+        reason,
+        approval_status,
+        issuer,
+        issuer_comment,
+        created_time,
+        approved_time,
+        settlement_status,
+        settlement_status_time,
+        settlement_reason_code,
+        settlement_response_text,
+        _line_number,
+        currency_code,
+        instance,
+        extract_filename,
+        ts,
+        littlepay_export_ts,
+        littlepay_export_date,
+        _content_hash,
+        _key,
+        _payments_key,
+        'refunds' AS source_table
+
+    FROM {{ ref('stg_littlepay__refunds') }}
+
+    -- should this be here, and should it also include 'AWAITING' if so?
+    WHERE approval_status != 'REFUSED'
+    QUALIFY DENSE_RANK() OVER (PARTITION BY refund_id ORDER BY littlepay_export_ts DESC) = 1
+
+),
+
+refunds_union AS (
+    SELECT *
+    FROM micropayments_table_refunds
+
+    UNION ALL
+
+    SELECT *
+    FROM refunds_table_refunds
+),
+
+int_payments__refunds AS (
+
+    SELECT
+
+        aggregation_id,
+        micropayment_id,
+        participant_id,
+        customer_id,
+        refund_amount,
+        transaction_date,
+        refund_id,
+        settlement_id,
+        retrieval_reference_number,
+        transaction_amount,
+        proposed_amount,
+        status,
+        initiator,
+        reason,
+        approval_status,
+        issuer,
+        issuer_comment,
+        created_time,
+        approved_time,
+        settlement_status,
+        settlement_status_time,
+        settlement_reason_code,
+        settlement_response_text,
+        _line_number,
+        currency_code,
+        instance,
+        extract_filename,
+        ts,
+        littlepay_export_ts,
+        littlepay_export_date,
+        _content_hash,
+        _key,
+        _payments_key,
+        source_table
+
+    FROM refunds_union
+
+    -- for refunds that appear in both micropayments and refunds source tables,
+    -- is this a functional way to only keep duplicates that are in refunds source table?
```

Comment:
I think that the number of refunds in this situation is like 3, so I would actually just manually drop them (`where _key NOT IN ([list their keys])`) from the micropayment refunds CTE above rather than doing this here